code/modules/clothing/spacesuits/hardsuit.dm

```diff
@@ -77,14 +77,13 @@
 
 /obj/item/clothing/head/helmet/space/hardsuit/process()
 	if(!rad_count)
-		grace_count++
-		if(grace_count == 2)
+		grace_count += SSOBJ_DT
+		if(grace_count >= 4)
 			soundloop.last_radiation = 0
 		return
 
 	grace_count = 0
-	rad_record -= rad_record/5
-	rad_record += rad_count/5
+	rad_record = (SSOBJ_DT / (8 + SSOBJ_DT)) * rad_count + (1 - SSOBJ_DT / (8 + SSOBJ_DT)) * rad_record // https://en.wikipedia.org/wiki/Low-pass_filter#Simple_infinite_impulse_response_filter
```

Comment:
Make a macro for this instead and use that for the other Geiger counter as well

Reply:
the old code is basically just `MC_AVERAGE()`

The new code? fuck if i know.

Given the intention of this, I don't think you need to scale the math here, but if you did, the multiplication form would be more readable.

`rad_record = (rad_count * (0.20 * (SSOBJ_DT/2))) + (rad_record * (0.80 * (SSOBJ_DT/2)))`

Reply:
It's a low pass filter, which is basically a moving average filter. I realise that this new form looks awful so I will make some macro that makes it more readable.